src/apps/lwaftr/icmp.lua

```diff
@@ -93,6 +94,22 @@ function new_icmpv4_packet(from_eth, to_eth, from_ip, to_ip, initial_pkt, l2_siz
    return new_pkt
 end
 
+function is_icmpv4(pkt)
+   return rd16(pkt.data + constants.o_ethernet_ethertype) == constants.n_ethertype_ipv4
```

Comment:
There's `is_ipv4` and `get_ipv4_proto`  defined at lwaftr.lua. At this moment these function are local. I prefer to reuse code as much as we can, so bugs are easier to fix (single point of failure). Ideally functions like this could be at `lwutil.lua` or other helper file. If it's easier just delocalize them and reuse them.